Source/Core/DolphinQt/GCMemcardManager.cpp

```diff
@@ -189,66 +217,62 @@ void GCMemcardManager::UpdateSlotTable(int slot)
 {
   m_slot_active_icons[slot].clear();
   m_slot_table[slot]->clear();
-  m_slot_table[slot]->setColumnCount(6);
+  m_slot_table[slot]->setColumnCount(COLUMN_COUNT);
   m_slot_table[slot]->verticalHeader()->setDefaultSectionSize(ROW_HEIGHT);
-  m_slot_table[slot]->verticalHeader()->setDefaultSectionSize(QHeaderView::Fixed);
   m_slot_table[slot]->setHorizontalHeaderLabels(
-      {tr("Banner"), tr("Title"), tr("Comment"), tr("Icon"), tr("Blocks"), tr("First Block")});
+      {tr("Banner"), tr("Title"), tr("Icon"), tr("Blocks")});
 
   if (m_slot_memcard[slot] == nullptr)
     return;
 
   auto& memcard = m_slot_memcard[slot];
   auto* table = m_slot_table[slot];
 
-  const auto create_item = [](const QString& string = {}) {
-    QTableWidgetItem* item = new QTableWidgetItem(string);
-    item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
-    return item;
-  };
-
   const u8 num_files = memcard->GetNumFiles();
+  const u8 free_files = Memcard::DIRLEN - num_files;
+  const u16 free_blocks = memcard->GetFreeBlocks();
   m_slot_active_icons[slot].reserve(num_files);
+  table->setRowCount(num_files);
   for (int i = 0; i < num_files; i++)
   {
-    int file_index = memcard->GetFileIndex(i);
-    table->setRowCount(i + 1);
+    const u8 file_index = memcard->GetFileIndex(i);
 
     const auto file_comments = memcard->GetSaveComments(file_index);
+    const u16 block_count = memcard->DEntry_BlockCount(file_index);
 
-    QString title;
-    QString comment;
-    if (file_comments)
-    {
-      title = QString::fromStdString(file_comments->first);
-      comment = QString::fromStdString(file_comments->second);
-    }
+    const QString title =
+        file_comments ? QString::fromStdString(file_comments->first).trimmed() : QString();
+    const QString comment =
+        file_comments ? QString::fromStdString(file_comments->second).trimmed() : QString();
+    auto banner = GetBannerFromSaveFile(file_index, slot);
+    auto icon_data = GetIconFromSaveFile(file_index, slot);
 
-    QString blocks = QStringLiteral("%1").arg(memcard->DEntry_BlockCount(file_index));
-    QString block_count = QStringLiteral("%1").arg(memcard->DEntry_FirstBlock(file_index));
+    auto* item_banner = new QTableWidgetItem();
+    auto* item_text = new QTableWidgetItem(QStringLiteral("%1\n%2").arg(title).arg(comment));
+    auto* item_icon = new QTableWidgetItem();
+    auto* item_blocks = new QTableWidgetItem(QStringLiteral("%1").arg(block_count));
```

Comment:
Given there's no particular formatting other than number->string needing to be done here, this can be simplified to

```suggestion
    auto* item_blocks = new QTableWidgetItem(QString::number(block_count));
```